lightning/kv/session.go

```diff
@@ -0,0 +1,225 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kv
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/pingcap/parser/mysql"
+	"github.com/pingcap/tidb/kv"
+	"github.com/pingcap/tidb/owner"
+	"github.com/pingcap/tidb/sessionctx/variable"
+	"github.com/pingcap/tidb/types"
+	"github.com/pingcap/tidb/util"
+	"github.com/pingcap/tidb/util/kvcache"
+	kvec "github.com/pingcap/tidb/util/kvencoder"
+	binlog "github.com/pingcap/tipb/go-binlog"
+)
+
+// transaction is a trimmed down Transaction type which only supports adding a
+// new KV pair.
+type transaction struct {
+	kvPairs []kvec.KvPair
```

Comment:
It is better to compound `kv.Transaction` interface here to avoid `panic("unexpected ***() call")`.